scripts/helm_redeploy_all_users.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env sh
+
+ENVIRONMENT=""
+HELM_CHART="config-user"
+HELM_CHART_VERSION="0.2.10"
+
+while true; do 
+	read -p "Enter the environment, i.e. dev or alpha, and press [Enter]: " environment
+   	case $environment in
+		[dev]*	) ENVIRONMENT=dev; break;;
+		[alpha]*) ENVIRONMENT=alpha; break;;
+		*	) echo "Only dev or alpha are acceptable answers"; exit;;
+	esac
+done
+
+CHART="../../analytics-platform-helm-charts/charts/${HELM_CHART}"
+RSTUDIO_CHART="../../analytics-platform-helm-charts/charts/rstudio"
+JUPYTER_CHART="../../analytics-platform-helm-charts/charts/jupyter-lab"
+RSTUDIO_CONFIG_FILE="../../analytics-platform-config/chart-env-config/${ENVIRONMENT}/rstudio.yml"
+JUPYTER_CONFIG_FILE="../../analytics-platform-config/chart-env-config/${ENVIRONMENT}/jupyter.yml"
+# USERNAME=$(helm list --max 10000 | grep "${HELM_CHART}" | awk '{print $1}' | sed "s/${HELM_CHART}-//g")
+USERNAME="dhirajnarwanimoj"
+USERNAME_FILE=$PWD/usernames.txt
+
+stat $CHART || exit
+
+declare -a USERNAMES=$USERNAME
+
+username_file() {
+
+	for i in $USERNAMES; do 
+		echo ${i[@]} >> $USERNAME_FILE
+		shift
+	done
+}
+
+username_file
+
+redeploy() {
+
+	set -x
+
+	while IFS= read -r u; do
+		
+		kubectl delete deployment -l app=jupyter-lab -n user-$u
```

Comment:
Maybe explain why you need to delete these deployments before upgrading their charts. I'd assume this would not be necessary, so it would be good to understand the reasoning.

Reply:
Yes because when i update the helm chart it will create new pods and not delete the old ones

Reply:
Really, upgrading a helm chart doesn't delete its previous deployments?! Either it's just me that didn't know this, or a snarky code comment along this line is needed ;-)

Reply:
@dhirajnarwanimoj the `--recreate-pods` flag may help with this: https://helm.sh/docs/intro/using_helm/#helpful-options-for-installupgraderollback